plugin/db/driver.go

```diff
@@ -420,8 +431,14 @@ type Driver interface {
 	// Dump and restore
 	// Dump the database, if dbName is empty, then dump all databases.
 	Dump(ctx context.Context, database string, out io.Writer, schemaOnly bool) error
-	// Restore the database from sc.
+	// Restore the database from sc, which is a full backup.
 	Restore(ctx context.Context, sc *bufio.Scanner) error
+	// RestoreIncremental restores the database using incremental backup in time range of [config.Start, config.End).
+	RestoreIncremental(ctx context.Context, config RecoveryConfig) error
+	// RestorePITR is a wrapper for restore a full backup and a range of incremental backup
+	RestorePITR(ctx context.Context, fullBackup *bufio.Scanner, config RecoveryConfig, database string, timestamp int64) error
```

Comment:
I don't feel these three interface makes perfect sense right now. It's okay to keep it for now but it might be better to have a separate package for doing MySQL restore. For example, RestoreIncremental() needs to know where the binary logs are stored. This needs coordination between Bytebase metadata. And I'm not sure the same interface applies to other database types. So it's better to find a separate package for this. And I think this will happen sooner than we expected.

Reply:
Actually, I have thought about it and worried that different databases would use different parameters for doing a PITR restore.
But I haven't thought about an appropriate package for them now.